### ISSUE TITLE
BatchNormalization axis default

### ISSUE THREAD

**Issue by valtron** [fchollet/keras#5328]
The default `axis` for mode 0 is -1. In the tf convention this is normalization per feature map, but in the (more common?) th convention, this normalizes per last dimension of the input. If input is >1D, this doesn't make sense: it treats one of the dimensions as "special".

Should the default be changed to conditionally set it to -1/+1 based on `image_dim_ordering`?

**Comment by valtron** [comment]
Bump so it doesn't get closed.

**Comment by fchollet** [comment]
This should definitely be closed.

Axis `-1 ` is always the features axis, *except* image tensors with `data_format="channels_first"`, which is very niche. The default should *not* depend on `image_data_format` because that only makes sense in the case of BN over image tensor.

**Comment by StefanoD** [comment]
Channel First is not a niche, if you want maximum performance on graphic cards.
[Torch](https://github.com/torch/image/blob/master/doc/saveload.md) for example, uses channel first.
